### DISCUSSION TITLE
How to limit amount of PRs per package?

### DISCUSSION THREAD

**Discussion by buffcode** [renovatebot/renovate#15002] [Request Help] [answered]
We have a dependency that is far behind, eg. v6 and current major is v14.

`separateMultipleMajor: true` is set, because we do not want to go `6 -> 14` but `6 -> 7 -> 8 -> maybe skip 9 and go directly to 10 -> 11 -> 14`.

So currently Renovate creates 7 PRs for this dependency and keeps them updated. Is there a way to limit the updates to a certain amount of opened PRs for this package (eg. maximum of 2 major releases)? I know I could use `allowedVersions` but then someone would also have to track what is configured to not accidentally miss a update.

**Comment by HonkingGoose** [comment, marked as answer]
Have you considered using the "Dependency Dashboard Approval" workflow [^dep-approval] for the package?

Extra config you'd need:

```json

{
  "packageRules": [
    {
      "matchPackagePatterns": ["^packagename"],
      "matchUpdateTypes": ["major"],
      "dependencyDashboardApproval": true
    }
  ]
}
```

After applying this config you have a Dependency Dashboard issue. You can now request `major` type updates for this package when you're ready. This way you only have the one major PR instead of seven. 😉

The `matchUpdateTypes` is set to `major` so you'll still get automatic PRs for all lower ranked updates. So you'll still get updates for the version of the package that you already use.

[^dep-approval]: https://docs.renovatebot.com/configuration-options/#dependencydashboardapproval

**Reply by buffcode** [reply]
Thanks, that's really a good idea. We already have the dashboard enabled, but used it rather sparingly so far.

**Reply by rarkins** [reply]
I have the same recommendation. Use approval for all major updates - in fact we do the same in this repo
